pkg/extgrpc/client.go

```diff
@@ -20,6 +20,27 @@ import (
 	"github.com/thanos-io/thanos/pkg/tracing"
 )
 
+// EndpointGroupGRPCOpts creates gRPC dial options for connecting to endpoint groups.
+// For details on retry capabilities, see https://github.com/grpc/proposal/blob/master/A6-client-retries.md#retry-policy-capabilities
+func EndpointGroupGRPCOpts() []grpc.DialOption {
+	serviceConfig := `
+{
+  "loadBalancingPolicy":"round_robin",
+  "retryPolicy": {
```

Comment:
Perhaps this should be configurable via flags?

Reply:
Yes, this could be a good idea. The new endpoint config would also help here.

Reply:
Can we add this flag in this PR, with this particular config as the default? 🙂

Reply:
This policy is really an implementation detail, I don't think it makes sense to expose it to the end user. The retry always has to be there so that the client will re-resolve dns endpoints when the one it's connected to goes away. This is how HA is implemented in gRPC.

Reply:
I see!